### PR TITLE
Fix brocken navx

### PR DESCRIPTION
**Pull request by fizzlebert**

### CODE REVIEW

File: utilities/navx.py [hunk 1]
```diff
@@ -1,43 +1,35 @@
...
-    @classmethod
-    def create_spi(cls, port=wpilib.SPI.Port.kMXP, spi_bitrate=None, update_rate_hz=None):
-        """Constructs the AHRS class using SPI communication, overriding the
-        default update rate with a custom rate which may be from 4 to 100,
-        representing the number of updates per second sent by the sensor.
-
-        This constructor allows the specification of a custom SPI bitrate, in bits/second.
-
-        .. note:: Increasing the update rate may increase the CPU utilization.
+        return math.radians(-self.ahrs.getRate()) * 100
```
Comment (auscompgeek) [inline utilities/navx.py]: Could you please note that this is a hack/approximation only?

**Comment by auscompgeek** [comment]
I just realised, we should actually document what this wrapper is for.

File: utilities/navx.py [hunk 2]
```diff
@@ -1,43 +1,41 @@
...
-import wpilib
 from robotpy_ext.common_drivers.navx.ahrs import AHRS
-from robotpy_ext.common_drivers.navx.registerio_spi import RegisterIO_SPI
+from wpilib.interfaces import PIDSource
 
-from utilities.functions import constrain_angle
 
+class NavX:
+    """Wrapper of robotpy navx to be inline with BNO055 api.
+    """
```
Comment (auscompgeek) [inline utilities/navx.py]: D200 One-line docstring should fit on one line with quotes

https://www.python.org/dev/peps/pep-0008/#documentation-strings
https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

Also, this should be capitalised correctly, and have correct grammar.

Also, it's _our_ BNO055 API, not just some random's off the internet. 😛

File: utilities/navx.py [hunk 3]
```diff
@@ -1,43 +1,41 @@
...
     def getHeadingRate(self):
-        return math.radians(-self.getRate())
-
-    # fix broken create_spi
-    @classmethod
-    def create_spi(cls, port=wpilib.SPI.Port.kMXP, spi_bitrate=None, update_rate_hz=None):
-        """Constructs the AHRS class using SPI communication, overriding the
-        default update rate with a custom rate which may be from 4 to 100,
-        representing the number of updates per second sent by the sensor.
+        return math.radians(-self.ahrs.getRate()) * 100  # short-term hack
```
Comment (auscompgeek) [inline utilities/navx.py]: This should mention _why_ this hack exists, not just that it's a hack.

I'm not sure how long you think "short-term" is either :)

File: utilities/navx.py [hunk 4]
```diff
@@ -1,43 +1,41 @@
...
+    PIDSourceType = PIDSource.PIDSourceType
+
+    def __init__(self):
+        self.ahrs = AHRS.create_spi(update_rate_hz=100)
+        self.pidsource = self.PIDSourceType.kDisplacement
 
     def getAngle(self):
-        raw = super().getAngle()
-        return constrain_angle(-math.radians(raw))
+        """Get navx angle
```
Comment (auscompgeek) [inline utilities/navx.py]: D205 1 blank line required between summary line and description
D400 First line should end with a period

File: utilities/navx.py [hunk 5]
```diff
@@ -1,43 +1,41 @@
...
+        returns:
```
Comment (auscompgeek) [inline utilities/navx.py]: I'm guessing this is meant to be following Google's style guide on docstrings? Because… this currently doesn't.

**Review by ArthurAllshire — APPROVED** [review]
.

**Pull request merged**